hibernate-core/src/main/java/org/hibernate/query/hql/internal/StandardHqlTranslator.java

```diff
@@ -43,6 +46,16 @@
  */
 public class StandardHqlTranslator implements HqlTranslator {
 
+	private static final ANTLRErrorListener PRETTIFY_ERROR_LISTENER = new BaseErrorListener() {
+		@Override
+		public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol, int line, int charPositionInLine, String msg, RecognitionException e) {
+			CommonTokenStream commonTokenStream = (CommonTokenStream) recognizer.getInputStream();
+			CharStream charStream = commonTokenStream.getTokenSource().getInputStream();
+			String hql = charStream.getText( Interval.of( 0, charStream.size() - 1 ) );
```

Comment:
The two parameters of `Interval#of(int,int)` are both inclusive.

A simplified way is simply: `charStream.toString()`, but its Javadoc never mentioned this so not sure whether it is future proof, so the current `CharStream#getText(Interval)` was chosen instead.

Reply:
It is not related to `offendingSymbol`. It is only related to `hql` which was the sole reason we need to create such listener each time when LL parsing step needs to do its due dilligence. The gist of this PR is to get the `hql` from the listener parameter instead so a single global listener or strategy impl will work in all cases. So we don't need to create an error listener instance each time LL step is run, saving resource and improve perf.

Reply:
If you think it too minor, I agree. Feel free to close it and the ticket. But I do think saving an unnecessary listener creation is worthwhile for such important logic flow.

Reply:
As far as I understand, the purpose of the proposed change is to avoid creating an object, right. But this code:

```java
CommonTokenStream commonTokenStream = (CommonTokenStream) recognizer.getInputStream();
CharStream charStream = commonTokenStream.getTokenSource().getInputStream();
String hql = charStream.getText( Interval.of( 0, charStream.size() - 1 ) );
```

Looks to me like it creates *at least one* object, and possibly more.

1. `Interval.of()` certainly does instantiation
2. `getText()` probably does instantiation too

So if anything this code is likely worse in terms of object instantiation, no?

Or am I misunderstanding something?

Reply:
you understand perfectly, and I think I missed that. I think we could close this PR and ticket then. How do you think?

Reply:
OK